core/src/main/java/com/linecorp/armeria/server/annotation/Params.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import com.linecorp.armeria.server.RouteBuilder;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(ParamsGroup.class)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface Params {
```

Comment:
We should not use a plural form here because we only specify a single parameter. However, we are already using `@Param` for injecting parameters, so I guess in this case we should name it differently to avoid confusion. e.g. `@RequiredParam` ?

Reply:
Thanks! `@Required...` sounds like that the parameter or header must exist. But we have *!=* case as well. So I changed it `@ConditionalParam` and `@ConditionalHeader`. What do you think? :-)